src/units/unit.cpp

```diff
@@ -1798,10 +1798,22 @@ int unit::resistance_against(const std::string& damage_name,bool attacker,const
 	if(!resistance_abilities.empty()) {
 		unit_abilities::effect resist_effect(resistance_abilities, 100-res);
 
-		res = 100 - std::min<int>(
-			resist_effect.get_composite_value(),
-			resistance_abilities.highest("max_value").first
-		);
+		std::optional<int> max_value;
+		unit_ability_list resistance_max_value;
+		resistance_max_value.append_if(resistance_abilities, [&](const unit_ability& i) {
+			return (*i.ability_cfg).has_attribute("max_value");
+		});
+		if(!resistance_max_value.empty()){
+			max_value = resistance_max_value.highest("max_value").first;
+		}
+		if(max_value){
+			res = 100 - std::min<int>(
+				resist_effect.get_composite_value(),
+				*max_value
+			);
+		} else {
+			res = 100 - resist_effect.get_composite_value();
+		}
```

Comment:
This could be simplified to:
```
if(!resistance_max_value.empty()){
	res = 100 - std::min<int>(
		resist_effect.get_composite_value(),
		resistance_max_value.highest("max_value").first
	);
} else {
	res = 100 - resist_effect.get_composite_value();
}
```
`std::optional<int> max_value;` can then be removed.

Reply:
it's done